relay-sampling/src/lib.rs

```diff
@@ -745,6 +749,9 @@ pub struct DynamicSamplingContext {
     /// Set on transaction start, or via `scope.transaction`.
     #[serde(default)]
     pub transaction: Option<String>,
+    /// The same rate with which this trace was sampled. This is a float between 0 and 1.
+    #[serde(default)]
+    pub sample_rate: Option<JsonStringifiedValue<f64>>,
```

Comment:
Do we need an error boundary here? Or is it OK to fail the entire `trace` header instead?

Reply:
I don't really know right now which behavior is best. if the sample rate is malformed like that, we will disable traces sampling, which we also do on other invalid fields

Reply:
OK that sounds fine.